maestrowf/interfaces/script/localpooladapter.py

```diff
@@ -0,0 +1,204 @@
+
+        return SubmissionRecord(submit_status, ret_code, job_id)
+
+    def check_jobs(self, joblist):
+        """
+        For the given job list, query execution status.
+
+        This method uses the scontrol show job <jobid> command and does a
+        regex search for job information.
+
+        :param joblist: A list of job identifiers to be queried.
+        :returns: The return code of the status query, and a dictionary of job
+                  identifiers to their status.
+        """
+        LOGGER.debug("Joblist type -- %s", type(joblist))
+        LOGGER.debug("Joblist contents -- %s", joblist)
+        if not joblist:
+            LOGGER.debug("Empty job list specified.")
+            return JobStatusCode.OK, {}
+        if not isinstance(joblist, list):
+            LOGGER.debug("Specified parameter is not a list.")
+            if isinstance(joblist, int):
+                LOGGER.debug("Integer found.")
+                joblist = [joblist]
+            else:
+                LOGGER.debug("Unknown type. Returning an error.")
+                return JobStatusCode.ERROR, {}
+
+        status = {jid: State.UNKNOWN for jid in joblist}
+        try:
+            for job_id, state in self._pool.get_all_status():
+                if job_id in status:
+                    status[job_id] = self._state(state)
+            chk_status = JobStatusCode.OK
+
+        except Exception as exception:
+            LOGGER.error(str(exception))
+            chk_status = JobStatusCode.ERROR
+
+        return chk_status, status
+
+    def cancel_jobs(self, joblist):
+        """
+        For the given job list, cancel each job.
+
+        :param joblist: A list of job identifiers to be cancelled.
+        :returns: The return code to indicate if jobs were cancelled.
+        """
+        # If we don"t have any jobs to check, just return status OK.
+        if not joblist:
+            return CancelCode.OK
+
+        c_status = set()
+        for job in joblist:
+            c_status.add(self._pool.cancel(job))
+
+        c_return = CancelCode.OK
+        ret_code = 0
+        if ExecCancel.FAILED in c_status:
+            c_return = CancelCode.ERROR
+            ret_code = -1
+        elif ExecCancel.JOBNOTFOUND in c_status:
+            ret_code = 1
+
+        return CancellationRecord(c_return, ret_code)
+
+    def _state(self, executor_state):
+        """
+        Map a scheduler specific job state to a Study.State enum.
+
+        :param executor_state: Enum representation of scheduler job status.
+        :returns: A Study.State enum corresponding to parameter job_state.
+        """
+        if executor_state == ExecTaskState.PENDING:
+            return State.PENDING
+        elif executor_state == ExecTaskState.RUNNING:
+            return State.RUNNING
+        elif executor_state == ExecTaskState.INITIALIZED:
+            return State.INITIALIZED
+        elif executor_state == ExecTaskState.CANCELLED:
+            return State.CANCELLED
```

Comment:
What about a CLOSING or CANCELLING status?  I've noticed at least on slurm that if you check it again too quick after cancelling (sometimes up to a minute) you'll get the CG status from the scheduler while it's still cleaning up/shutting it down.  Or can pending capture this?

Reply:
In the current implementation, those don't exist. The pool technically will cancel a job (in this case a process) outright.